### PR TITLE
add benchmark CL

### LINKED ISSUE #4: feature request: auto benchmark a CL
It'd be nice to be able to invoke compilecmp with just a CL number: `compilecmp -cl 205777`. It should grab the CL and use the CL's parent commit as before and the CL's commit as after.

https://github.com/bradfitz/gitutil/tree/master/git-clpatch has some goo to help interact with gerrit. This is even easier and nicer since we don't actually need to touch our working state at all (all the compilecmp work happens in worktrees). Just a question of setting the refs correctly in this bit of code:

```go
switch flag.NArg() {
	case 0:
	case 1:
		beforeRef = flag.Arg(0)
	case 2:
		beforeRef = flag.Arg(0)
		afterRef = flag.Arg(1)
	default:
		log.Fatal("usage: compilecmp [before-git-ref] [after-git-ref]")
	}
```

cc @cuonglm

### PR DESCRIPTION
**Pull request by cuonglm**
Fixes #4

### CODE REVIEW

File: main.go [hunk 1]
```diff
@@ -26,6 +28,7 @@ var (
 	flagCount = flag.Int("n", 15, "iterations")
 	flag386   = flag.Bool("386", false, "run in 386 mode")
 	flagEach  = flag.Bool("each", false, "run for every commit between before and after")
+	flagCl    = flag.String("cl", "", "run benchmark on CL number")
```
Comment (josharian) [inline main.go]: flag.CL please. Maybe flag.Int instead of flag.String?

File: main.go [hunk 2]
```diff
@@ -44,13 +47,25 @@ func main() {
 	flag.Parse()
 	beforeRef := "master"
 	afterRef := "HEAD"
+	if flagCl != nil {
```
Comment (josharian) [inline main.go]: Do `if flag.NArg() > 0 { log.Fatal("-cl NNN is incompatible with ref arguments") }` here. Then you can remove the flagCL check below. To actually correctly compare a CL against a commit that isn't one of its parents, you need to rebase/cherry-pick, and that is definitely beyond the scope of this script.

File: main.go [hunk 3]
```diff
@@ -44,13 +47,25 @@ func main() {
...
+		clHead, parent, err := clHeadAndParent(*flagCl)
+		if err != nil {
+			log.Fatalf("failed to get CL %s information", *flagCl)
```
Comment (josharian) [inline main.go]: Please print the error here too.

File: main.go [hunk 4]
```diff
@@ -44,13 +47,25 @@ func main() {
...
+		}
+		if parent != "" {
```
Comment (josharian) [inline main.go]: I don't see how `parent == ""` can happen with the Go repo. No-one is going to create a new disconnected branch. Instead, if `parent == ""`, call `log.Fatal`.

**Review by josharian — CHANGES_REQUESTED** [review]
Thanks! I'm excited to have this.

File: main.go (hunk 1 above)
Comment (cuonglm) [reply main.go]: Fixed

File: main.go (hunk 2 above)
Comment (cuonglm) [reply main.go]: Fixed

File: main.go (hunk 3 above)
Comment (cuonglm) [reply main.go]: Fixed

File: main.go (hunk 4 above)
Comment (cuonglm) [reply main.go]: Fixed

**Comment by josharian** [comment]
Thanks!

**Comment by josharian** [comment]
I've used this a few times already. It's very nice to have. Thanks again.

**Pull request merged**